SPDX-FileCopyrightText: 2026 ukoOS Contributors

SPDX-License-Identifier: BSD-2-Clause

diff --git c/include/sbi/riscv_locks.h w/include/sbi/riscv_locks.h
index 492935f5..ae771d1f 100644
--- c/include/sbi/riscv_locks.h
+++ w/include/sbi/riscv_locks.h
@@ -34,7 +34,7 @@ typedef struct {
 #define DEFINE_SPIN_LOCK(x)	\
 	spinlock_t SPIN_LOCK_INIT(x)
 
-int spin_lock_check(spinlock_t *lock);
+bool spin_lock_check(spinlock_t *lock);
 
 int spin_trylock(spinlock_t *lock);
 
diff --git c/include/sbi/sbi_types.h w/include/sbi/sbi_types.h
index 38e3565b..ba74b9ce 100644
--- c/include/sbi/sbi_types.h
+++ w/include/sbi/sbi_types.h
@@ -44,7 +44,11 @@ typedef unsigned long long	uint64_t;
 #error "Unexpected __riscv_xlen"
 #endif
 
-typedef int			bool;
+#if __STDC_VERSION__ < 202311L
+typedef _Bool			bool;
+#define true			1
+#define false			0
+#endif
 typedef unsigned long		ulong;
 typedef unsigned long		uintptr_t;
 typedef unsigned long		size_t;
@@ -54,10 +58,8 @@ typedef unsigned long		virtual_size_t;
 typedef unsigned long		physical_addr_t;
 typedef unsigned long		physical_size_t;
 
-#define TRUE			1
-#define FALSE			0
-#define true			TRUE
-#define false			FALSE
+#define TRUE			true
+#define FALSE			false
 
 #define NULL			((void *)0)
 
