--- a/include/configs/cv181x-asic.h
+++ b/include/configs/cv181x-asic.h
@@ -171,18 +171,6 @@
 /* #define CONFIG_USB_DWC2_REG_ADDR	0x04340000 */
 /* Enable below CONFIG for fastboot */
 
-/* To use usb fastboot, you need to enable below Kconfig
- * CONFIG_USB, CONFIG_USB_GADGET and CONFIG_USB_GADGET_DWC2_OTG
- */
-
-#ifdef CONFIG_USB_GADGET
-#define CONFIG_USB_FUNCTION_FASTBOOT
-#define CONFIG_USB_GADGET_DOWNLOAD
-#define CONFIG_G_DNL_MANUFACTURER "Cvitek"
-#define CONFIG_G_DNL_VENDOR_NUM   0x18d1
-#define CONFIG_G_DNL_PRODUCT_NUM 0x4ee0
-#endif
-
 #define CONFIG_IPADDR			192.168.0.3
 #define CONFIG_NETMASK			255.255.255.0
 #define CONFIG_GATEWAYIP		192.168.0.11
